pilot/pkg/model/service.go

```diff
@@ -846,6 +847,23 @@ func (u NoopAmbientIndexes) WorkloadsForWaypoint(scope WaypointScope) []*Workloa
 
 var _ AmbientIndexes = NoopAmbientIndexes{}
 
+type AddressInfo struct {
+	*workloadapi.Address
+}
+
+func (i AddressInfo) ResourceName() string {
+	var name string
+	switch addr := i.Type.(type) {
+	case *workloadapi.Address_Workload:
+		ii, _ := netip.AddrFromSlice(addr.Workload.Address)
+		name = addr.Workload.Network + "/" + ii.String()
```

Comment:
that method only exists for the wrapper WorkloadInfo object, the base Workload object does not have the method

Reply:
can we leave a TODO here to change the primary key to UID? per the design doc, the primary xds key should be UID and secondary keys for lookup are any of the `network/IP` pairs (since there will eventually be support for many addresses for dual stack). This part of the design doc has not been implemented yet, and my vote would be to punt for now to keep these PRs as manageable as possible